enterprise/models.py

```diff
@@ -2513,78 +2515,149 @@ class DefaultEnterpriseEnrollmentIntention(TimeStampedModel, SoftDeletableModel)
         """
-        The enrollment deadline for this course.
+        The enrollment deadline for the course run.
         """
         return datetime.datetime.min
 
+    def determine_content_type(self):
+        """
+        Determines the content_type for a given content_key by validating the return value
+        from `content_metadata_for_content_key`. First determines if the configured content_key
+        matches the returned key, then checks if it matches any of the returned course runs.
+        
+        Returns either COURSE, COURSE_RUN, or None (if neither can be determined).
+        """
+        content_metadata = self.content_metadata_for_content_key
+        if not content_metadata:
+            return None
+
+        # Determine whether the returned key matches the configured content_key and
+        # the returned metadata denotes the content type as a course.
+        content_metadata_key = content_metadata.get('key', '')
+        content_metadata_content_type = content_metadata.get('content_type', '')
+        if content_metadata_key.lower() == self.content_key.lower() and content_metadata_content_type == self.COURSE:
+            return self.COURSE
+
+        # Determine if the content_key matches any of the course runs
+        # in the content metadata.
+        course_runs = content_metadata.get('course_runs', [])
+        course_run = next(
+            (course_run for course_run in course_runs if course_run['key'].lower() == self.content_key.lower()),
+            None
+        )
+        return self.COURSE_RUN if course_run is not None else None
+
     def clean(self):
-        if not (self.content_type or self.get_content_type):
-            raise ValidationError('The content key did match a valid course or course-run')
+        """
+        Raise ValidationError if no course run or content type exists.
+        """
+        super().clean()
+
+        existing_record = DefaultEnterpriseEnrollmentIntention.all_objects.filter(
+            enterprise_customer=self.enterprise_customer,
+            content_key=self.content_key,
+        ).exclude(uuid=self.uuid).first()
+
+        if existing_record and existing_record.is_removed:
+            existing_record_admin_url = reverse(
+                'admin:enterprise_defaultenterpriseenrollmentintention_change',
+                args=[existing_record.uuid],
+            )
+            message = _(
+                f'A default enrollment intention with this enterprise customer and '
+                f'content key already exists, but is soft-deleted. Please restore '
+                f'it <a href="{existing_record_admin_url}">here</a>.',
+            )
+            raise ValidationError({
+                'content_key': mark_safe(message)
+            })
+
+        if not self.course_run:
+            raise ValidationError({
+                'content_key': _('The content key did not resolve to a valid course run.')
+            })
```

Comment:
[nit] As part of our async conversation, this check is also an inferred check for the `content_type` since both calls will fail (`self.course_run` and `self.determine_content_type()`)

[optional] I would add a comment about how this check is also an inferred check on the `content_type`